src/plugins/smith_waterman/src/SmithWatermanAlgorithm.cpp

```diff
@@ -414,7 +414,7 @@ void SmithWatermanAlgorithm::calculateMatrixForAnnotationsResult() {
     }
 #endif
 
-    free(matrix);
+    delete matrix;
```

Comment:
delete [] matrix

Reply:
> Just replacing malloc calls with new is not enough. There is a solution offered in the comment to the issue.

This task is called from https://github.com/ugeneunipro/ugene/blob/master/src/corelibs/U2Private/src/TaskSchedulerImpl.cpp#L1052-L1057
```C
        try {
            ti->task->run();
            assert(ti->task->getState() == Task::State_Running);
        } catch (const std::bad_alloc &) {
            onBadAlloc(ti->task);
        }
```
Calling `new` throws `bad_alloc` in case of low memory (unlike `malloc()`) which is caught here and sets an error in `onBadAlloc`

Reply:
I agree with @sgrekhov . There is no need to write error to the operationStatus member. If there is not enough memory to allocate, "new" throws the "bad_alloc" exception and TaskScheduler catches it.

Reply:
Ok, you convinced me. But fix delete calls, pointers should be released as pointers to array, not as pointers to a value.